base/files/file_util_starboard.cc

```diff
@@ -145,58 +148,59 @@ bool AbsolutePath(FilePath* path) {
   return path->IsAbsolute();
 }
 
-bool DeleteFile(const FilePath &path, bool recursive) {
-  internal::AssertBlockingAllowed();
-  const char *path_str = path.value().c_str();
-  struct ::stat info;
-  bool directory = ::stat(path_str, &info) == 0 && S_ISDIR(info.st_mode);
-  if (!recursive || !directory) {
-    return SbFileDelete(path_str);
+// Borrowed from file_util_posix.cc
+bool DoDeleteFile(const FilePath& path, bool recursive) {
+  ScopedBlockingCall scoped_blocking_call(FROM_HERE, BlockingType::MAY_BLOCK);
+  // Reset errno in the beginning of the funciton
+  errno = 0;
```

Comment:
reset errno in the beginning of the function, otherwise raspi might reports lingering errno here.